src/UWP/ArcGISRuntime.UWP.Viewer/Helpers/ArcGISLoginPrompt.cs

```diff
@@ -0,0 +1,117 @@
+                {
+                    // Use the OAuth implicit grant flow
+                    GenerateTokenOptions = new GenerateTokenOptions
+                    {
+                        TokenAuthenticationType = TokenAuthenticationType.OAuthAuthorizationCode
+                    },
+
+                    // Indicate the url (portal) to authenticate with (ArcGIS Online)
+                    ServiceUri = new Uri(ArcGISOnlineUrl)
+                };
+
+                // Call GetCredentialAsync on the AuthenticationManager to invoke the challenge handler
+                Credential cred = await AuthenticationManager.Current.GetCredentialAsync(challengeRequest, false);
+                loggedIn = cred != null;
+            }
+            catch (OperationCanceledException)
+            {
+                // OAuth login was canceled, no need to display error to user.
+            }
+            catch (Exception ex)
+            {
+                // Login failure
+                await new MessageDialog(ex.Message, "Login failed").ShowAsync();
+            }
+
+            return loggedIn;
+        }
+
+        // ChallengeHandler function that will be called whenever access to a secured resource is attempted
+        public static async Task<Credential> PromptCredentialAsync(CredentialRequestInfo info)
+        {
+            Credential credential = null;
+
+            try
+            {
+                // IOAuthAuthorizeHandler will challenge the user for OAuth credentials
+                credential = await AuthenticationManager.Current.GenerateCredentialAsync(info.ServiceUri);
+            }
+            catch (OperationCanceledException)
+            {
+                // OAuth login was canceled, no need to display error to user.
+            }
+
+            return credential;
+        }
+
+        public static void SetChallengeHandler()
+        {
+            // Define the server information for ArcGIS Online
+            ServerInfo portalServerInfo = new ServerInfo(new Uri(ArcGISOnlineUrl))
+            {
+                TokenAuthenticationType = TokenAuthenticationType.OAuthAuthorizationCode,
+                OAuthClientInfo = new OAuthClientInfo(AppClientId, new Uri(OAuthRedirectUrl))
+            };
+
+            // If a client secret has been configured, set the authentication type to OAuth client credentials.
+            if (!string.IsNullOrEmpty(ClientSecret))
+            {
+                // Use OAuthClientCredentials if you need a refresh token (and have specified a valid client secret).
+                portalServerInfo.TokenAuthenticationType = TokenAuthenticationType.OAuthClientCredentials;
+                portalServerInfo.OAuthClientInfo.ClientSecret = ClientSecret;
+            }
+
+            // Get a reference to the (singleton) AuthenticationManager for the app
+            AuthenticationManager thisAuthenticationManager = AuthenticationManager.Current;
```

Comment:
I'd remove this in favor of using `AuthenticationManager.Current` directly below.